libsolidity/formal/SMTChecker.cpp

```diff
@@ -740,10 +727,15 @@ void SMTChecker::initializeLocalVariables(FunctionDefinition const& _function)
 
 void SMTChecker::resetStateVariables()
 {
-	for (auto const& variable: m_stateVariables)
+	for (auto const& variable: m_variables)
 	{
-		newValue(*variable.first);
-		setUnknownValue(*variable.first);
+		VariableDeclaration const* _decl = dynamic_cast<VariableDeclaration const*>(variable.first);
+		solAssert(_decl, "");
```

Comment:
Could also use `dynamic_cast<VariableDeclaration const&>(...)` instead. It will throw a different exception, but I guess that is still fine.